buildSrc/src/main/java/org/elasticsearch/gradle/LoggedExec.java

```diff
@@ -47,11 +50,16 @@
  * A wrapper around gradle's Exec task to capture output and log on error.
  */
 @SuppressWarnings("unchecked")
-public class LoggedExec extends Exec {
+public class LoggedExec extends Exec implements FileSystemOperationsAware {
 
     private static final Logger LOGGER = Logging.getLogger(LoggedExec.class);
     private Consumer<Logger> outputLogger;
 
+    @Inject
+    protected FileSystemOperations getFileSystemOperations() {
+        throw new UnsupportedOperationException();
```

Comment:
Can't we inject this into the constructor as well? If so, doesn't that avoid having this weird exception throwing method implementation?

Reply:
done